src/DynamicDiscreteSamplers.jl

```diff
@@ -24,9 +24,6 @@ An object that confomrs the the `Weights` interface and cannot be resized.
 """
 struct FixedSizeWeights <: Weights
     m::Memory{UInt64}
-    global _FixedSizeWeights
-    _FixedSizeWeights(m::Memory{UInt64}) = new(m)
-    FixedSizeWeights(len::Integer) = new(initialize_empty(Int(len)))
```

Comment:
The point of these restricted inner constructors (and the omission of the default inner constructor) is that if we allowed the default inner constructor `FixedSizeWeights(m::Memory)` would be public API (it's a public function called on a public type) but would produce garbage unless the memory conforms to the private memory representation.

Further, it would break

```julia-repl
julia> FixedSizeWeights(Memory{Float64}([1,2,3]))
3-element FixedSizeWeights:
 1.0
 2.0
 3.0
```

Because the memory `[1,2,3]` would be interpreted as the internal memory, not as a collection of weights.

Reply:
right, then I will restructure this PR to use inner constructors instead, trying to add the least redundancy in doing so. 

another question: what is the reason for `_FixedSizeWeights`?

Reply:
Last commit should resolve this issue